doc/prscheduler.md

```diff
@@ -0,0 +1,155 @@
+![open_close_pr.jpg](images/open_close_pr.jpg)
+
+* config.yml is a configuration file which must contain the following information:
+```
+# cat config.yml
+repository:
+  owner: $repo_owner
+  name: $repo_name
+credentials:
+  token: $github_token
+```
+
+
+### Ansible playbook
+
+#### Use the playbook
+
+In order to install and configure the scheduler tool on a machine, you can
+use the ansible playbook `ansible/prepare_openclose_pr_tool.yml`. You need
+first to edit the inventory file `ansible/hosts/runners` and define the
+machine that will be used as scheduler (the playbook will apply to all
+ the machines in the [scheduler] group):
+```
+# cat ansible/hosts/runners
+[scheduler]
+1.2.3.4        # the scheduler IP
+```
+
+Then run:
+```
+ansible-playbook -i ansible/hosts/runners ansible/prepare_openclose_pr_tool.yml
+```
+
+The playbook will prompt you for variables:
+```
+# ansible-playbook -i ansible/hosts/runners ansible/prepare_openclose_pr_tool.yml
+Provide an unique identifier for the PR: testing_master    # $identifier
+Branch name to open PR against it: master                  # $branch
+Repo owner to create branches: freeipa-pr-ci               # $repo_owner
+GitHub token: 123qwerty                                    # $github_token
+Provide a path to a .freeipa-pr-ci.yaml file to use it to open new PRs: ipatests/prci_definitions/nightly_master
+Provide a ssh key that has push access to the git repo: ~/path/to/ssh_key #$git_sshkey
+Should the PR be open against the upstream repo?. Type no for opening it agaist your own freeipa repo (yes/no): no
+```
+
+You can run the ansible playbook multiple times to schedule multiple PRs,
+but to each PR you want to create you need to provide a different
+identifier (ID). This ID is used to create the branch and to name the
+PR.
+
+
+#### Playbook tasks
+
+The playbook is performing multiple tasks in order to install and configure
+the python script:
+* installation of python and deps for ansible modules (python2, python2-dnf,
+libselinux-python)
+* prompting of required variables (identifier, branch, repo_owner,
+github_token, prci_config, git_sshkey, pr_against_upstream)
+* installation of the role `automation/openclose_pr`:
+  * installation of required packages: python3-github3py, PyYAML, git
+  * installation of python dependencies: CacheControl, gitpython
+  * creation of the base repository where the tool and config is stored:
+`/root/openclose_pr`
+  * creation of the configuration file `/root/openclose_pr/config.yaml`
+containing repo info and credentials
+  * installation of the python script in `/root/openclose_pr/openclose_pr.py`
+  * creation of the test definition file in
+`/root/openclose_pr/.freeipa-pr-ci.yaml`
+  * preparation of ssh configuration in order to access github.com
+  * creation of a clone repo from $repo_owner/freeipa.git to
+`/root/openclose_pr/freeipa/`
+  * configuration of the cron job to run daily at 23:00
+
+#### Current scheduling
```

Comment:
I would again mention here that we are usually opening nightly PRs from "freeipa-pr-ci" GitHub user.

Reply:
Good point, updated.